src/tools/perf/lib/libperf.c

```diff
@@ -1186,22 +1212,37 @@ static ucs_status_t ucp_perf_test_setup_endpoints(ucx_perf_context_t *perf,
                                                   uint64_t features)
 {
     ucs_status_t status;
+    unsigned group_index;
+    unsigned group_size;
+    unsigned peer_index;
     unsigned i;
 
+    group_size  = rte_call(perf, group_size);
+
+    if ((group_size != 2) &&
+            !(perf->params.flags & UCX_PERF_TEST_FLAG_LOOPBACK)) {
```

Comment:
if loopback, expect that grou_size would be ==1

Reply:
not clear here, the original code catches the issue when gr. size is not 2, 1 for example, I said in loopback case this is not an issue

Reply:
Do u mean I should check in loopback case if it exactly 1 ?

Reply:
> Do u mean I should check in loopback case if it exactly 1 ?

Yes